lineage/src/main/java/org/apache/gravitino/lineage/sink/LineageSinkManager.java

```diff
@@ -19,23 +19,88 @@
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
 import io.openlineage.server.OpenLineage.RunEvent;
 import java.io.Closeable;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.gravitino.lineage.LineageConfig;
+import org.apache.gravitino.listener.EventBus;
+import org.apache.gravitino.listener.EventListenerManager;
+import org.apache.gravitino.listener.api.event.EventWrapper;
 
-@SuppressWarnings("unused")
 public class LineageSinkManager implements Closeable {
+  private EventBus eventBus;
+  private final EventListenerManager eventListenerManager;
 
-  public void initialize(List<String> sinks, Map<String, String> LineageConfigs) {}
+  public LineageSinkManager() {
+    this.eventListenerManager = new EventListenerManager();
+  }
+
+  public void initialize(List<String> sinks, Map<String, String> LineageConfigs) {
+    Map<String, String> eventListenerConfigs =
+        transformToEventListenerConfigs(sinks, LineageConfigs);
+    eventListenerManager.init(eventListenerConfigs);
+    this.eventBus = eventListenerManager.createEventBus();
+    eventListenerManager.start();
+  }
 
   // Checks if the sink queue size surpasses the threshold to avoid overwhelming lineage sinks.
   public boolean isHighWatermark() {
-    return false;
+    return eventBus.isHighWatermark();
   }
 
-  public void sink(RunEvent runEvent) {}
+  @VisibleForTesting
+  static Map<String, String> transformToEventListenerConfigs(
+      List<String> sinks, Map<String, String> lineageConfigs) {
+    Map<String, String> eventListenerConfigs = new HashMap<>();
+    eventListenerConfigs.putAll(generateEventListenerConfigs(sinks, lineageConfigs));
+    eventListenerConfigs.putAll(lineageConfigs);
+    return eventListenerConfigs;
+  }
+
+  private static Map<String, String> generateEventListenerConfigs(
+      List<String> sinks, Map<String, String> lineageConfigs) {
+    HashMap eventListenerConfigs = new HashMap();
+    if (sinks.isEmpty()) {
+      return eventListenerConfigs;
+    }
+
+    String queueCapacity = lineageConfigs.get(LineageConfig.LINEAGE_SINK_QUEUE_CAPACITY);
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(queueCapacity), "Lineage sink queue capacity is not set");
+    int capacityPerSink = Integer.valueOf(queueCapacity) / sinks.size();
+
+    eventListenerConfigs.put(
+        EventListenerManager.GRAVITINO_EVENT_LISTENER_NAMES, String.join(",", sinks));
+    sinks.forEach(
+        sinkName -> {
+          eventListenerConfigs.put(
+              sinkName + "." + EventListenerManager.GRAVITINO_EVENT_LISTENER_CLASS,
+              LineageSinkEventListener.class.getName());
+          eventListenerConfigs.put(
+              sinkName + "." + EventListenerManager.GRAVITINO_EVENT_LISTENER_QUEUE_CAPACITY,
+              String.valueOf(capacityPerSink));
+        });
+    return eventListenerConfigs;
```

Comment:
There's some logic between Lineage sink and event listeners, also including some transform logic, can you add some comments in the code to explain the details?

Reply:
updated the comment